README.md

```diff
@@ -0,0 +1,52 @@
+# Introduction
+
+The **hal_gigadevice** is a set of standard firmware library from GigaDevice.
+
+# How submit code
+
+ - When submiting a newer firmware library version, it is important make sure
+that the last firmware version will be merged. The library version can be
+checked at [GigaDevice MCU Site](http://www.gd32mcu.com/en/download/).
+ - This repository is managed mainly on Linux. This means that changes should
+ be submited using Linux LF format. Usually running a similar command to
+ `find * type f -print0 | xargs -0 dos2unix -k` should be enougth.
+ - Make sure directory structure is respected.
+
+## Directory Structure
+
+### ARM
+
+```
+.
+└── GD32XXX
+    ├── CMSIS
+    │   └── GD
+    │       └── GD32XXX
+    └── standard_peripheral
+        ├── Include
+        └── Source
+```
+
+### RISC-V
+
+```
+└── GD32VF103
+    ├── RISCV
+    │   ├── drivers
+    │   ├── env_Eclipse
+    │   ├── env_IAR
+    │   └── stubs
+    └── standard_peripheral
+        ├── Include
+        └── Source
+```
+
+
+ The contribution should follow
+ [Zephyr Contribution Guidelines](https://docs.zephyrproject.org/latest/contribute/index.html).
+
+## Know issues
+
+ - The `timer_init` function name conflicts with Zephyr. This should be renamed
+ to `timer_initialize`, see [ddde23c](https://github.com/zephyrproject-rtos/hal_gigadevice/commit/ddde23cadb1f089263fae2d167e4d83a742d373c).
+ - Some CAN defines conflicts with Zephyr, see [223ccbf](https://github.com/zephyrproject-rtos/hal_gigadevice/commit/223ccbfd8ae4bb6b3ada14e4a44995baeefad762) as guide.
```

Comment:
Better create a list of issues and their equivalents, these commits may not contain all changes made in future revisions. If we come up with a standardized way of patching collisions, I can write a script to automate this.

```suggestion
GD32 HAL does not use any namespace or prefix, therefore collisions are easy. Names that collide with
existing Zephyr definitions or functions are prefixed with a double underscore, e.g. `__colliding_function()`.
Below you will find a list of collisions and their equivalents. When adding or updating
a new HAL library you should patch all the names listed below.

- `timer_init`: `__timer_init`
- etc.
```

Reply:
I think we should also stop using `gd32xxx_libopt.h`, since it pollutes almost every single file (it gets included everywhere). An approach like this https://github.com/zephyrproject-rtos/hal_stm32/tree/main/scripts/genllheaders would make more sense: have generic hal headers that get included where they are needed, probably reducing the chances of collision.

Reply:
For the name conflict, there have another case like BIT, which have same implement with Zephyr.

Reply:
> For the name conflict, there have another case like BIT, which have same implement with Zephyr.

It's not ideal, but I guess it is acceptable in this case since it's a trivial macro. Still I think we need to ensure that Zephyr `BIT` is always included first, so it is always used.